### ISSUE TITLE
Empty city/state for few records

### ISSUE THREAD

**Issue by MartinMikita** [OSMNames/OSMNames#50]
I have found few records with empty `city` and `state` values, while display name contains everything.

For example:

```
{
 'id': 15341364,
 'alternative_names': "",
 'boundingbox': [13.736494, 50.10368, 13.738399, 50.105],
 'city': "",
 'class': "highway",
 'country': "Czech Republic",
 'country_code': "cz",
 'county': "Středočeský kraj",
 'display_name': "Hálkova, Rakovník, okres Rakovník, Středočeský kraj, Czech Republic",
 'importance': 0.1,
 'lat': 50.104366,
 'lon': 13.737429,
 'name': "Hálkova",
 'name_suffix': "CZ",
 'osm_id': "30985694",
 'osm_type': "way",
 'place_rank': 26.0,
 'rank': 2680111.75,
 'state': "",
 'street': "Hálkova",
 'type': "residential",
 'wikidata': "",
 'wikipedia': ""}

{
 'id': 15334559,
 'alternative_names': "",
 'boundingbox': [14.163356, 49.786163, 14.172256, 49.788288],
 'city': "",
 'class': "highway",
 'country': "Czech Republic",
 'country_code': "cz",
 'county': "Středočeský kraj",
 'display_name': "Hálkova, Dobříš, okres Příbram, Středočeský kraj, Czech Republic",
 'importance': 0.1,
 'lat': 49.787533,
 'lon': 14.167943,
 'name': "Hálkova",
 'name_suffix': "CZ",
 'osm_id': "25218558",
 'osm_type': "way",
 'place_rank': 26.0,
 'rank': 2680111.75,
 'state': "",
 'street': "Hálkova",
 'type': "residential",
 'wikidata': "",
 'wikipedia': ""}
```

Other data with same class/type contains values.
Is it possible to fix this problem?

**Comment by klokan** [comment]
I am afraid, that if you need this within the next month - you may better find out an algorithm how to extract the information from the existing display_name.

**Comment by and01** [comment]
The function `getParentInfo` does the matching of city/county/state as well as the display name. These fields are determined the following way:

- if the parent has rank 16 -> city
- if the parent has rank 12 -> county
- if the parent has rank 8 -> state

Maybe it would be better to adapt it like this

- if the parent has rank 16 < 20 -> city
- if the parent has rank 12 < 16 -> county
- if the parent has rank 8 < 12 -> state

**Comment by simran001** [comment]
Solved with PR #82
